src/system/ExtensionsModule/Controller/ModuleController.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Copyright Zikula Foundation 2016 - Zikula Application Framework
+ *
+ * This work is contributed to the Zikula Foundation under one or more
+ * Contributor Agreements and licensed to You under the following license:
+ *
+ * @license GNU/LGPLv3 (or at your option, any later version).
+ *
+ * Please see the NOTICE file distributed with this source code for further
+ * information regarding copyright and licensing.
+ */
+
+namespace Zikula\ExtensionsModule\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Zikula\Component\SortableColumns\Column;
+use Zikula\Component\SortableColumns\SortableColumns;
+use Zikula\Core\Controller\AbstractController;
+use Zikula\ExtensionsModule\ExtensionEvents;
+use Zikula\ThemeModule\Engine\Annotation\Theme;
+
+/**
+ * Class ModuleController
+ * @package Zikula\ExtensionsModule\Controller
+ * @Route("/module")
+ */
+class ModuleController extends AbstractController
+{
+    /**
+     * @Route("/list/{pos}")
+     * @Theme("admin")
+     * @Template
+     * @param Request $request
+     * @param int $pos
+     * @return array
+     */
+    public function viewModuleListAction(Request $request, $pos = 1)
+    {
+        if (!$this->hasPermission('ZikulaExtensionsModule::', '::', ACCESS_ADMIN)) {
+            throw new AccessDeniedException();
+        }
+
+        $sortableColumns = new SortableColumns($this->get('router'), 'zikulaextensionsmodule_module_viewmodulelist');
+        $sortableColumns->addColumns([new Column('displayname'), new Column('state')]);
+        $sortableColumns->setOrderByFromRequest($request);
+
+        $vetoEvent = new GenericEvent();
+        $this->get('event_dispatcher')->dispatch(ExtensionEvents::REGENERATE_VETO, $vetoEvent);
+        if (!$vetoEvent->isPropagationStopped() & $pos == 1) {
+            // regenerate the extension list only when viewing the first page
+            $extensionsInFileSystem = $this->get('zikula_extensions_module.bundle_sync_helper')->scanForBundles();
+            $this->get('zikula_extensions_module.bundle_sync_helper')->syncExtensions($extensionsInFileSystem);
+        }
+
```

Comment:
no. As I said above, Multisites would use an event handler to veto the regeneration of modules.